server/Order.js

```diff
@@ -0,0 +1,158 @@
+    }
+
+    parseItems(itemsList) { // who parses itemsJSON into Items?
+        itemsList.forEach(item => {
+            this.inventoryItems[item.name] = item
+        });
+    }
+
+    // Beginning of Observable's duties
+    notifyFoodBank(foodBankURL) {
+        try {
+            // let fbServer = new XMLHttpRequest();
+            // fbServer.open("POST", foodBankURL, true);
+            // fbServer.setRequestHeader('Content-Type', 'application/json');
+            // fbServer.send(JSON.stringify({
+            //     "orderId": this.orderId,
+            //     "status": this.progressStatus
+            // }))
+
+            // fbServer.onreadystatechange = function () {
+            //     if (this.status === 200) {
+            //         console.log("200");
+            //     }
+            // }
+
+            // fbServer.send();
+            console.log("200")
+
+        } catch (error) {
+            console.log("Can't notify entity.")
+        }
+    }
+
+    notifyGroceryStore(groceryStoreURL) {
+        try {
+            // let gsServer = new XMLHttpRequest();
+            // gsServer.open("POST", groceryStoreURL, true);
+            // gsServer.setRequestHeader('Content-Type', 'application/json');
+            // gsServer.send(JSON.stringify({
+            //     "orderId": this.orderId,
+            //     "status": this.progressStatus
+            // }))
+
+            // gsServer.onreadystatechange = function () {
+            //     if (this.status === 200) {
+            //         console.log("200");
+            //     }
+            // }
+
+            // gsServer.send();
+            console.log("200");
+        } catch (error) {
+            console.log("Can't notify entity.")
+        }
+    }
+
+
+    notifyDrivers(context) { // push notification
+        if (this.driverId.length !== 0) {
+            this.driverId.forEach(driver => {
+                driver.messages = context;
+            });
+            console.log('Driver notified with ' + context);
+        } else {
+            console.log("Notification failed" + error);
+        }
+    }
+
+    addDriver(driver) { // adds observer
+        if (this.isDriverValid(driver)) {
+            this.driverId[driver.orderId] = driver;
+            console.log("Driver added.")
+        } else {
+            console.log("Can't assign driver" + error);
+        }
+    }
+
+    removeDriver(driverId) { // removes observer
+        if (driverId in this.driverId) {
+            delete this.driverId[driverId];
+            console.log("Driver" + driverId + " removed.");
+        } else {
+            console.log("Can't unsubscribe driver " + error);
+        }
+    }
+
+    isDriverValid(driver) {
+        //check if Driver instance has ID, name, reg. vehicle
+
+        return ((driver.id !== null) && driver.name !== undefined && driver.vehicle !== undefined);
+    }
+    // End of Observable duties
+
+    // Beginning of Observer duties
+    updateStatus(newStatus) {
```

Comment:
Create status enum with the following states:
Received
Invalid
Looking For Driver
In Progress
Picked Up
Delivered